app/src/main/java/bg/zahov/app/ui/home/HomeViewModel.kt

```diff
@@ -25,45 +29,56 @@ import java.time.LocalDateTime
 import java.time.format.DateTimeFormatter
 import java.time.temporal.TemporalAdjusters
 
+data class HomeUiState(
+    val isLoading: Boolean = false,
+    val username: String = "",
+    val numberOfWorkouts: String = "",
+    val barData: BarData = BarData(),
+    val isChartLoading: Boolean = true
+)
+
+data class BarData(
+    var chartVisibility: Int = View.GONE,
+    var xMin: Float = 0f,
```

Comment:
Data classes should use `val` types instead of `var`. If something needs to be changed you either make a copy of the instance or it should not be part of the data class.

Reply:
This is still a `var`.